.github/workflows/tests_linux.yml

```diff
@@ -97,7 +108,16 @@ jobs:
         uses: actions/checkout@v3
         with:
           path: main
-          fetch-depth: 2
+          fetch-depth: ${{ inputs.lightning-version == 'stable' && 0 || 2 }}
```

Comment:
What is `&& 0 || 2` selecting for?

Reply:
I'm not 100% sure, I took that from @rashidnhm . I think the logic is: if latest fetch depth == 2, if stable fetch depth == 0 (i.e. everything) to get all tags. We could instead use the default fetch depth (1) everywhere and use the `fetch-tags` option to get tags where needed.